src/ranking.jl

```diff
@@ -171,3 +171,178 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(v::AbstractVector, value; method=:inc, sorted=false)
+
+Compute the quantile(s)-position in [0-1] of a `value` relative to a collection `v`, e.g. 
+a quantile rank of x means that x% of the elements in `v` are lesser or lesser-equal the 
+given `value`. 
+
+The keyword argument `method` (default `:inc`) can be:
+
+`:inc` - The inverse of the quantile based in def. 7 in Hyndman and Fan (1996). 
```

Comment:
would it be possible to give definitions here apart from the references (for all options?)

Reply:
While researching the different percentrank methods, I had difficulty finding clear definitions in the documentation for each implementation, in each language. The best documentation i've found is for percentile methods, but percentrank is pretty sketchy.

For example, see PercentRank in [Excel](https://support.microsoft.com/en-us/office/percentrank-function-f1b5836c-9619-4847-9fc9-080ec9024442), in [MariaDB](https://mariadb.com/kb/en/percent_rank/) and in [Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.percentileofscore.html). 

So, I tried my best to give a minimum valid reference for the user understand the options in the method arg.
If you have any suggestion, feel free to give us. We would really appreciate.

Reply:
> If you have any suggestion, feel free to give us. We would really appreciate.

I understand your concerns. However, please try to understand me. My responsibility is to review the PR if it correctly implements something. In order to do this I need to have a formal definition and check if the code correctly represents it. Currently it is very hard. If I have a hard time doing this then probably any user who will want to use this function later will also have the same problem.

What we could do (after we finalize the implementation) is do the reverse: when we have implementation we can write down what it does exactly (I can help with this if you want after the implementation part of the PR is finished).

Thank you for working on this.

Reply:
>I understand your concerns. However, please try to understand me. My responsibility is to review the PR if it correctly implements something. In order to do this I need to have a formal definition and check if the code correctly represents it.

I tottally understand you. 
So far I've only found general definitions of percentrank and I haven't found clear definitions for each method. 
I've done this definition search in the past, I didn't find anything about it, but I'll redo it to try to find formal definitions for each method 
Later, in this week, I come back to say if I find something useful


> What we could do (after we finalize the implementation) is do the reverse: when we have implementation we can write down what it does exactly (I can help with this if you want after the implementation part of the PR is finished).

That ideia sounds good to me. I'm open to follow your instructions when we start to write down what it does exactly.

Reply:
@AugustoCL - I would just define the five variables that we compute (they are pretty natural in interpretation) and then verbally how they are used the computations we have in the code. Eg. for `:inc` method the computations are https://github.com/JuliaStats/StatsBase.jl/pull/741/files#diff-2497cd36f6671b22f5e43d99f95a9e84d391ac51c207975b6f346a60efe95562R275-R286

Of course we can also keep the references.

Reply:
I did what you suggested in commit 6132df5. Now it's ready for your review.